SwitchSpeak-iOS/SwitchSpeak-iOS/CrumbManager.swift

```diff
@@ -18,22 +18,30 @@ class CrumbStack {
         // The UI element holding the string describing the breadcrumb.
         // i.e. "I", "am", etc.
         var label:UILabel
+        // scale is used to decide the size of crumb according to the number of crumbs
+        var scale:CGFloat
         
         init(content:String) {
             label = UILabel()
             label.text = content
+            scale = 1
         }
         
         /*
          * Sets the color of the breadcrumb as well as its location on the screen
          * relative to the view containing it. The provided index corresponds
          * to the crumb's index in the stack holding it.
          */
-        func setLocationAndColor(byIndex index:Int) {
-            label.frame = CGRect(x: 10 + (index % 5) * 80, y: 40 * (index / 5 + 1), width: 80, height: 40)
+        
+        func setLocationAndSize(byIndex index:Int, withWidth width:CGFloat) {
+            label.frame = CGRect(x: 10 + CGFloat(index) * width  * scale, y: 40, width: width * scale, height: 40)
+        }
+        
+        func setLocationAndColor(byIndex ind:Int, withWidth width:CGFloat) {
```

Comment:
Can you please make the index variable name consistent? I would be fine with either `ind` or `index`- up to you.